sdk/src/clients/client_v2.rs

```diff
@@ -0,0 +1,931 @@
+            send_messages_batch: None,
+            partitioner: None,
+            encryptor: None,
+            message_handler: None,
+            message_channel_sender: None,
+        }
+    }
+
+    /// Creates a new `IggyClientV2` with the provided client implementation for the specific transport and the optional configuration for sending and polling the messages in the background.
+    /// Additionally it allows to provide the custom implementations for the message handler, partitioner and encryptor.
+    pub fn create(
+        client: Box<dyn ClientV2>,
+        config: IggyClientBackgroundConfigV2,
+        message_handler: Option<Box<dyn MessageHandler>>,
+        partitioner: Option<Box<dyn Partitioner>>,
+        encryptor: Option<Box<dyn Encryptor>>,
+    ) -> Self {
+        if partitioner.is_some() {
+            info!("Partitioner is enabled.");
+        }
+        if encryptor.is_some() {
+            info!("Client-side encryption is enabled.");
+        }
+
+        let client = IggySharedMut::new(client);
+        let send_messages_batch = Arc::new(Mutex::new(SendMessagesBatch {
+            commands: VecDeque::new(),
+        }));
+        if config.send_messages.enabled && config.send_messages.interval > 0 {
+            info!("Messages will be sent in background.");
+            Self::send_messages_in_background(
+                config.send_messages.interval,
+                config.send_messages.max_messages,
+                client.clone(),
+                send_messages_batch.clone(),
+            );
+        }
+
+        IggyClientV2 {
+            client,
+            config: Some(config),
+            send_messages_batch: Some(send_messages_batch),
+            message_handler: message_handler.map(Arc::new),
+            message_channel_sender: None,
+            partitioner,
+            encryptor,
+        }
+    }
+
+    /// Returns the channel receiver for the messages which are polled in the background. This will only work if the `start_polling_messages` method is called.
+    pub fn subscribe_to_polled_messages(&mut self) -> Receiver<PolledMessage> {
+        let (sender, receiver) = flume::unbounded();
+        self.message_channel_sender = Some(Arc::new(sender));
+        receiver
+    }
+
+    /// Starts polling the messages in the background. It returns the `JoinHandle` which can be used to await for the completion of the task.
+    #[allow(clippy::too_many_arguments)]
+    pub fn start_polling_messages<F>(
+        &self,
+        stream_id: &Identifier,
+        topic_id: &Identifier,
+        partition_id: Option<u32>,
+        consumer: &Consumer,
+        strategy: &PollingStrategy,
+        count: u32,
+        on_message: Option<F>,
+        config_override: Option<PollMessagesConfig>,
+    ) -> JoinHandle<()>
+    where
+        F: Fn(PolledMessage) + Send + Sync + 'static,
+    {
+        let client = self.client.clone();
+        let mut interval = Duration::from_millis(100);
```

Comment:
Is this 100 a default value which is the same as in `PollMessagesConfig` struct default trait implementation? Perhaps some internal const value could be used to make it explicit and use in both places?

Reply:
Fixed :)